link-parent/link-ilp-over-http/src/test/java/org/interledger/link/http/MultilateralUrlFactoryTest.java

```diff
@@ -0,0 +1,43 @@
+package org.interledger.link.http;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.interledger.core.InterledgerAddress;
+
+import okhttp3.HttpUrl;
+import org.junit.Test;
+
+public class MultilateralUrlFactoryTest {
+
+  private static final InterledgerAddress OPERATOR = InterledgerAddress.of("g.operator");
+  private static final HttpUrl MULTILATERAL_URL = HttpUrl.parse("https://www.test.com/spsp-%/ilp");
+  private static final HttpUrl NON_MULTILATERAL_URL = HttpUrl.parse("https://www.test.com/spsp/ilp");
+
+  @Test
+  public void getOutgoingUrlWithMultilateralUrl() {
+    MultilateralUrlFactory factory = new MultilateralUrlFactory(() -> OPERATOR, MULTILATERAL_URL);
+    assertThat(factory.getOutgoingUrl(OPERATOR.with("spsp").with("0").with("secretstuff")))
+        .isEqualTo(HttpUrl.parse("https://www.test.com/spsp-0/ilp"));
+    assertThat(factory.getOutgoingUrl(OPERATOR.with("spsp").with("ONE").with("secretstuff")))
+        .isEqualTo(HttpUrl.parse("https://www.test.com/spsp-ONE/ilp"));
+  }
+
+  @Test
+  public void getOutgoingUrlWithNonMultilateral() {
+    MultilateralUrlFactory factory = new MultilateralUrlFactory(() -> OPERATOR, NON_MULTILATERAL_URL);
+    assertThat(factory.getOutgoingUrl(OPERATOR.with("spsp").with("0").with("secretstuff")))
+        .isEqualTo(NON_MULTILATERAL_URL);
+    assertThat(factory.getOutgoingUrl(OPERATOR.with("spsp").with("ONE").with("secretstuff")))
+        .isEqualTo(NON_MULTILATERAL_URL);
+  }
+
+  @Test
+  public void getOutgoingUrlWithNonMultilateralAddress() {
+    MultilateralUrlFactory factory = new MultilateralUrlFactory(() -> OPERATOR, MULTILATERAL_URL);
```

Comment:
Are we sure this is how we want this to function? Won't the Link blow up when it tries to resolve `https://www.test.com/spsp-%/ilp`?

Reply:
@nhartner and I discussed and we decided that this will end up in a REJECT anyway. If an infra operator wanted to handle URLs with a percent-sign like this, they could. 

If this becomes a pain-point, we could specify a `defaultURL` here that doesn't include a `%` wildcard so that if a destination address cannot be mapped, then at least the HTTP call might succeed. However, not going to do this now because it's ambiguous if anyone will need it.